bookshelf/books/firestore.js

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019, Google, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+// [START bookshelf_firestore_client]
+const Firestore = require('@google-cloud/firestore');
+
+const db = new Firestore();
+const collection = 'Book';
+
+// [END bookshelf_firestore_client]
+
+// Lists all books in the database sorted alphabetically by title.
+// The callback is invoked with ``(err, books, nextPageToken)``.
+function list(limit, token, cb) {
```

Comment:
What are your thoughts on going fully async here?

Reply:
I'll update this in a second pass.